addons/mail/models/discuss/discuss_channel.py

```diff
@@ -508,17 +506,39 @@ def _notify_get_recipients(self, message, msg_vals, **kwargs):
                 sql_query,
                 (email_from or '', list(pids), [author_id] if author_id else [], )
             )
-            for partner_id, partner_share, notif in self._cr.fetchall():
+            for partner_id, lang, partner_share, notif, ushare in self._cr.fetchall():
                 # ocn_client: will add partners to recipient recipient_data. more ocn notifications. We neeed to filter them maybe
                 recipients_data.append({
+                    'active': True,
+                    'groups': [],
                     'id': partner_id,
+                    'is_follower': False,
+                    'lang': lang,
+                    'notif': notif,
                     'share': partner_share,
-                    'active': True,
-                    'notif': notif or 'email',
                     'type': 'user' if not partner_share and notif else 'customer',
-                    'groups': [],
+                    'uid': False,
+                    'ushare': ushare,
                 })
 
+        if self.channel_type == 'chat':
```

Comment:
`if self.is_chat or self.channel_type == "group":` from `_notify_thread` turned into `if self.channel_type == 'chat':` only?

I'm not sure what we want functionally but my feeling says only `channel_type == "channel"` should be excluded from notification.

Reply:
I am unsure what we want, because the method was called two times, and as there were some shortcuts everywhere I am unsure original author really knew what was the expected result.  We could add 'group' if you think they want web push spam; but not sure we want whatsapp channels to send web push for example (currently they don't).

Reply:
Also: 'group' channel were only forcing calling "send push notifications", but members were only added for chat channels. Code is a bit vague.

Reply:
`is_chat` included `livechat` and `whatsapp`, and `group` was listed, so I think it was intended for all of them.

Only `channel` is typically removed from notification stuff as we consider them too spammy. And even for that, this is arguable as we now have the mute setting.

For me the only thing we light lack is the ability for a user to mute a full category, but otherwise there should be no restriction on notification for channel type, and just a choice of mute or not.

Reply:
In JS we notify (locally + open chat window) for all but `channel`.
https://github.com/odoo/odoo/pull/149943

Reply:
Ok. I think it is actually currently a bit broken due to the way the code is designed, it will be better now then :) .